File: go-simple-chat-app/message.go
package main

import "time"

type Message struct {
	UserID    int
	UserName  string
	Text      string
	Timestamp time.Time
}

func NewMessage(user User, text string) Message {
	return Message{
		UserID:    user.ID,
		UserName:  user.Name,
		Text:      text,
		Timestamp: time.Now(),
	}
}

File: rust-simple-chat-app/src/message.rs
use chrono::{DateTime, Local};

#[derive(Clone, Debug)]
pub struct Message {
    pub user_id: u32,
    pub user_name: String,
    pub text: String,
    pub timestamp: DateTime<Local>,
}

impl Message {
    pub fn new(user_id: u32, user_name: String, text: String) -> Self {
        Self {
            user_id,
            user_name,
            text,
            timestamp: Local::now(),
        }
    }

    pub fn display(&self) {
        println!(
            "[{}] User {} ({}) says: {}",
            self.timestamp.format("%Y-%m-%d %H:%M:%S"),
            self.user_id,
            self.user_name,
            self.text
        );
    }
}